storage/innobase/dict/dict0dict.cc

```diff
@@ -4094,9 +4094,11 @@ static void dict_table_persist_to_dd_table_buffer_low(dict_table_t *table) {
 
   size = dict_persist->persisters->write(metadata, buffer);
 
-  dberr_t error =
-      table_buffer->replace(table->id, table->version, buffer, size);
-  ut_a(error == DB_SUCCESS);
+  if (!srv_apply_log_only) {
+    dberr_t error =
+        table_buffer->replace(table->id, table->version, buffer, size);
+    ut_a(error == DB_SUCCESS);
+  }
```

Comment:
Add a comment on why we cant apply MLOG_TABLE_DYNAMIC_META records during recovery.